mautic/mautic
# Prevent exception when creating a new field when background processing is enabled

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | 3.3
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | yes
| Related user documentation PR URL      | 
| Related developer documentation PR URL | 
| Issue(s) addressed                     | 

#### Description:
https://github.com/mautic/mautic/pull/8828 introduced the ability to process custom fields in the background. However, the API endpoint did not handle the AbortColumnCreateException  exception resulting in the API request to 500 instead of being handled gracefully. 

This PR fixes that by returning a 202 response code (accepted) if the field is being created in the background. It'll return a 201 if it's created during the request. 

To support the functional tests, AbstractMauticTestCase.php had to be modified to allow tests to set their own config parameters (in this case, enable and disable background processing). 

It was also found while writing the functional tests that the API did not return all of the properties used by the field and so this also adds them. 

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Add `'create_custom_field_in_background' => true,` to the instance's config log 
2. Use the API to create a new custom field https://developer.mautic.org/#create-field
3. Note the 500 "uh oh" response
4. Disable `create_custom_field_in_background` and then try again with a new field. This time it will create with a 201 response. 

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Repeat the above. Instead of a 500 when create_custom_field_in_background is enabled, you should get a 202 response. Then when create_custom_field_in_background is disabled, you should get a 201 response.
